### PR TITLE
CAPZ: update Kubernetes upgrade tests with newer versions

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
Refresh k8s version upgrade tests:

- removes 1.19 -> 1.20 
- removes 1.20 -> 1.21
- adds 1.23 -> 1.24

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-main-upgrades.yaml [hunk 1]
```diff
@@ -69,13 +25,13 @@ periodics:
...
       env:
         - name: KUBERNETES_VERSION_UPGRADE_FROM
-          value: "stable-1.20"
-        - name: KUBERNETES_VERSION_UPGRADE_TO
           value: "stable-1.21"
+        - name: KUBERNETES_VERSION_UPGRADE_TO
+          value: "stable-1.22"
         - name: ETCD_VERSION_UPGRADE_TO
-          value: "3.4.13-0"
+          value: "3.5.1-0"
```
Comment (jackfrancis) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-main-upgrades.yaml:32]: We want to be careful about suggesting to users to use etcd 3.5, but we especially don't want to use 3.5.1. See:

https://groups.google.com/g/etcd-dev/c/sad9tgmKU7Y/m/vhArFVevBgAJ

Comment (CecileRobertMichon) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-main-upgrades.yaml]: updated to 3.5.3

**Review by jackfrancis — APPROVED** [review]
/lgtm
/approve

**Pull request merged**